src/main/scala/rise/elevate/rules/algorithmic.scala

```diff
@@ -275,7 +275,7 @@ object algorithmic {
     case expr @ App(generate(), Lambda(i, App(App(App(select(), t),
       App(App(map(), f), e1)),
       App(App(map(), g), e2))))
-    if e1 == e2 && !contains[Rise](i).apply(e1) =>
+    if e1 =~= e2 && !contains[Rise](i).apply(e1) =>
```

Comment:
more of these, we should only rewrite fully typed expressions.

Reply:
The original behavior of == was to accept type placeholders. I will try and see which ones I can get rid of though.